pontoon/contributors/static/js/settings.js

```diff
@@ -47,6 +47,123 @@ $(function () {
     });
   });
 
+  $('.generate-token-btn').click(function (e) {
+    e.preventDefault();
+
+    const csrfmiddlewaretoken = $('input[name="csrfmiddlewaretoken"]').val();
+
+    $.ajax({
+      url: '/generate-token/',
+      type: 'POST',
+      data: {
+        name: $('.token-name-input').val(),
+        csrfmiddlewaretoken: csrfmiddlewaretoken,
+      },
+      success: function (response) {
+        if (response.status === 'success') {
+          const newTokenHTML = `
+            <li data-token-id="${response.data['new_token_id']}">
+                <div class="token-card created controls clearfix">
+                <div class="token-header">
+                  <span class="token-name">${response.data['new_token_name']}</span>
+                  <div>
+                    <button class="button delete-btn far fa-trash-alt" tabindex="-1" data-token-id="${response.data['new_token_id']}"></button>
+                    <input type="checkbox">
+                  </div>
+                </div>
+                <div class="token-info-container">
+                  <div class="token-info">
+                    <span class="icon fas fa-calendar-alt"></span>
+                    <span>Expires on:</span> 
+                    <span class="date">${response.data['new_token_expires_at']}</span>
+                  </div>
+                  <div class="token-info">
+                    <span class="icon fas fa-calendar-alt"></span>
+                    <span>Last used:</span> 
+                    <span class="date">
+                      ${response.data['new_token_last_used'] ? response.data['new_token_last_used'] : 'Never'}
+                    </span>
+                  </div>
```

Comment:
No need to show the Last used info for the newly created token. It will always be "Never".